matlab_simulation/code v1.0/ME597-7-NonlinearSteering/angleWrap.m

```diff
@@ -0,0 +1,17 @@
+%% angleWrap
+% A function that receives an angle and binds it between [-pi, pi].
+%
+% Input:
+% angle - Any arbitray angle measurement in radians
+%
+% Output:
+% angle - The same angle except numerically bound from -pi to pi
+function [angle] = angleWrap(angle)
+
+while ( angle < -pi || angle > pi)
+    if (angle < -pi)
+        angle = angle + 2*pi;
+    else
+        angle = angle - 2*pi;
+    end
+end
```

Comment:
A minor thing, but I think it is slightly more efficient to split this into two consecutive while loops, that way you don't need the `if` statement or the `||` operation.